Yesod/Auth/OAuth2/Github.hs

```diff
@@ -77,27 +77,27 @@ oauth2GithubScoped clientId clientSecret scopes = basicPlugin {apDispatch = disp
                 , oauthCallback            = Nothing
                 }
 
-        withState state = authOAuth2 "github"
-            (oauth {oauthOAuthorizeEndpoint = oauthOAuthorizeEndpoint oauth `BS.append` "&state=" `BS.append` encodeUtf8 state})
-            fetchGithubProfile
+        -- withState state = authOAuth2 "github"
+        --     (oauth {oauthOAuthorizeEndpoint = oauthOAuthorizeEndpoint oauth `BS.append` "&state=" `BS.append` encodeUtf8 state})
+        --     fetchGithubProfile
 
         basicPlugin = authOAuth2 "github" oauth fetchGithubProfile
 
-        dispatch "GET" ["forward"] = do
-            state <- liftIO $ fmap (T.pack . toString) nextRandom
-            setSession "githubState" state
-            apDispatch (withState state) "GET" ["forward"]
-
-        dispatch "GET" ["callback"] = do
-            state <- lift $ runInputGet $ ireq textField "state"
-            savedState <- lookupSession "githubState"
-            _ <- apDispatch basicPlugin "GET" ["callback"]
-            case savedState of
-                Just saved | saved == state -> apDispatch basicPlugin "GET" ["callback"]
-                Just saved -> invalidArgs ["state: " `mappend` state `mappend` ", and not: " `mappend` saved]
-                _ -> invalidArgs ["state: " `mappend` state]
-
-        dispatch method ps = apDispatch basicPlugin method ps
+        -- dispatch "GET" ["forward"] = do
+        --     state <- liftIO $ fmap (T.pack . toString) nextRandom
+        --     setSession "githubState" state
+        --     apDispatch (withState state) "GET" ["forward"]
+
+        -- dispatch "GET" ["callback"] = do
+        --     state <- lift $ runInputGet $ ireq textField "state"
+        --     savedState <- lookupSession "githubState"
+        --     _ <- apDispatch basicPlugin "GET" ["callback"]
+        --     case savedState of
+        --         Just saved | saved == state -> apDispatch basicPlugin "GET" ["callback"]
+        --         Just saved -> invalidArgs ["state: " `mappend` state `mappend` ", and not: " `mappend` saved]
+        --         _ -> invalidArgs ["state: " `mappend` state]
+
+        -- dispatch method ps = apDispatch basicPlugin method ps
```

Comment:
This is awesome to see. Please go ahead and delete it, not need to keep commented code around.